kv/bolt/build_test.go

```diff
@@ -1,54 +1,133 @@
-package bolt_test
-
-// import (
-// 	"testing"
-
-// 	"github.com/google/go-cmp/cmp"
-// 	"github.com/influxdata/chronograf"
-// )
-
-// func
-// func TestBuildStore_Get(t *testing.T) {
-// 	type wants struct {
-// 		build *chronograf.BuildInfo
-// 		err   error
-// 	}
-// 	tests := []struct {
-// 		name  string
-// 		wants wants
-// 	}{
-// 		{
-// 			name: "When the build info is missing",
-// 			wants: wants{
-// 				build: &chronograf.BuildInfo{
-// 					Version: "pre-1.4.0.0",
-// 					Commit:  "",
-// 				},
-// 			},
-// 		},
-// 	}
-// 	for _, tt := range tests {
-// 		client, err := NewTestClient()
-// 		if err != nil {
-// 			t.Fatal(err)
-// 		}
-// 		if err := client.Open(context.TODO()); err != nil {
-// 			t.Fatal(err)
-// 		}
-// 		defer client.Close()
-
-// 		b := client.BuildStore
-// 		got, err := b.Get(context.Background())
-// 		if (tt.wants.err != nil) != (err != nil) {
-// 			t.Errorf("%q. BuildStore.Get() error = %v, wantErr %v", tt.name, err, tt.wants.err)
-// 			continue
-// 		}
-// 		if diff := cmp.Diff(got, tt.wants.build); diff != "" {
-// 			t.Errorf("%q. BuildStore.Get():\n-got/+want\ndiff %s", tt.name, diff)
-// 		}
-// 	}
-// }
-
-// func TestBuildStore_Update(t *testing.T) {
-
-// }
+package bolt
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/chronograf"
+)
+
+// TestNow is a set time for testing.
+var TestNow = time.Date(2000, time.January, 1, 0, 0, 0, 0, time.UTC)
+
+// TestClient wraps *bolt.Client.
+type TestClient struct {
+	Client *client
+	path   string
+}
+
+// NewTestClient creates new *bolt.Client with a set time and temp path.
+func NewTestClient() (*TestClient, error) {
+	f, err := ioutil.TempFile("", "chronograf-bolt-")
+	if err != nil {
+		return nil, errors.New("unable to open temporary boltdb file")
+	}
+	f.Close()
+
+	ctx := context.TODO()
+	b, err := NewClient(ctx,
+		WithPath(f.Name()),
+		WithNow(func() time.Time { return TestNow }),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &TestClient{
+		Client: b,
+		path:   f.Name(),
+	}, nil
+}
+
+func (c *TestClient) Close() error {
+	defer os.Remove(c.path)
+	return c.Client.Close()
+}
+
+func TestBuildStore_Get(t *testing.T) {
+	type wants struct {
+		build chronograf.BuildInfo
+		err   error
+	}
+	tests := []struct {
+		name  string
+		wants wants
+	}{
+		{
+			name: "When the build info is missing",
+			wants: wants{
+				build: chronograf.BuildInfo{
+					Version: "1.8.0",
```

Comment:
is this `1.8.0` going to need to be changed every time we update patch versions?

Reply:
This is really an arbitrary value that will likely never get used, I copied it from the old code and updated it.. but i think taking it out will be a better solution.

Reply:
looking at it again.. it's for some migration stuff so i'll leave it in. ...athough I stripped out a ton of migration code because there is a clear migration path; if people are still using a 2+ year old version of chronograf, they can upgrade to 1.7.x line and then 1.8.x. :thinking: